trieste/acquisition/rule.py

```diff
@@ -972,6 +1004,111 @@ def update(
         """
         ...
 
+    def _get_tags(self, tags: Set[Tag]) -> Tuple[Set[Tag], Set[Tag]]:
+        # Separate tags into local (matching index) and global tags (without matching
+        # local tag).
+        local_gtags = set()
+        global_tags = set()
+        for tag in tags:
+            ltag = LocalizedTag.from_tag(tag)
+            if not ltag.is_local:
+                global_tags.add(tag)
+            elif ltag.local_index == self.region_index:
+                local_gtags.add(ltag.global_tag)
+
+        # Only keep global tags that don't have a matching local tag.
+        global_tags = global_tags.difference(local_gtags)
+
+        return local_gtags, global_tags
+
+    def select_models(
+        self, models: Optional[Mapping[Tag, ProbabilisticModelType]]
+    ) -> Optional[Mapping[Tag, ProbabilisticModelType]]:
+        """
+        Select models belonging to this region for acquisition.
+
+        :param models: The model for each tag.
+        :return: The models belonging to this region.
+        """
+        if models is None:
+            _models = {}
+        elif self.region_index is None:
+            # If no index, then return the global models.
+            _models = {
+                tag: model
+                for tag, model in models.items()
+                if not LocalizedTag.from_tag(tag).is_local
+            }
+        else:
+            # Prefer matching local model for each tag, otherwise select the global model.
+            local_gtags, global_tags = self._get_tags(set(models))
+
+            _models = {}
+            for tag in local_gtags:
+                ltag = LocalizedTag(tag, self.region_index)
+                _models[ltag] = models[ltag]
+            for tag in global_tags:
+                _models[tag] = models[tag]
+
+        return _models if _models else None
+
+    def select_datasets(
```

Comment:
This is basically identical to select_models. Maybe make it generic?

```
def select_in_region(self, mapping: Optional[Mapping[Tag, T]]) -> Optional[Mapping[Tag, T]]:
    ....
```

and then you can pass in either `models` or `datasets` as required.